File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_moisture import Moisture

# Callback function for moisture value 
def cb_moisture(moisture):
    print('Moisture Value: ' + str(moisture))

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    m = Moisture(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Set Period for moisture callback to 1s (1000ms)
    # Note: The moisture callback is only called every second if the 
    #       moisture has changed since the last call!
    m.set_moisture_callback_period(1000)

    # Register moisture callback to function cb_moisture
    m.register_callback(m.CALLBACK_MOISTURE, cb_moisture)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletMoisture.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletMoisture;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

// Callback function for moisture value callback
function cb_moisture($moisture)
{
    echo "Moisture Value: $moisture\n";
}

$ipcon = new IPConnection(); // Create IP connection
$m = new BrickletMoisture(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register moisture value callback to function cb_moisture
$m->registerCallback(BrickletMoisture::CALLBACK_MOISTURE, 'cb_moisture');

// Set period for moisture value callback to 1s (1000ms)
// Note: The moisture value callback is only called every second
//       if the moisture value has changed since the last call!
$m->setMoistureCallbackPeriod(1000);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
